modules/mauth-authenticator-akka-http/src/main/scala/com/mdsol/mauth/akka/http/MAuthDirectives.scala

```diff
@@ -269,71 +218,66 @@ trait MAuthDirectives extends StrictLogging {
-          case None => {
-            val msg = s"$timeHeaderName header supplied with bad format: [$timeValue]"
-            logger.error(msg)
-            reject(MalformedHeaderRejection(headerName = timeHeaderName, errorMsg = msg, None))
-          }
-        }
-      } else {
-        reject(MissingHeaderRejection(timeHeaderName))
-      }
-    } else {
-      val msg = s"$authHeaderName header is not isPresent"
-      logger.error(msg)
-      if (isReject)
-        reject(MissingHeaderRejection(authHeaderName))
-      else
-        provide(new MauthHeaderValues("", 0L))
-    }
-  }
+  private def extractRequestHeader(request: HttpRequest, headerName: String): String =
+    request.headers
+      .find(_.name().compareToIgnoreCase(headerName) == 0)
+      .map(_.value())
+      .getOrElse("")
 
   /**
     * Extracts the authentication header value of the HTTP request header for latest version of Mauth
     *
-    * @param request HttpRequest
     * @param disableV1
     *        the flag to specify if Mauth V1 is disabled or not.
     *        If Mauth V1 is disabled, extracts the authentication header of MCC-Authentication only.
     *        Otherwise, extracts the authentication header of X-MWS-Authentication if MCC-Authentication header is not found.
     *
-    * @return Directive1[MauthHeaderValues] of Mauth authentication header values
+    * @return Directive1[MauthHeaderValues] of Mauth authentication header values for V1 or V2
     *         the request is rejected with a MissingHeaderRejection if the expected header is not present
     */
-  def extractLatestAuthenticationHeaders(request: HttpRequest, disableV1: Boolean): Directive1[MauthHeaderValues] = {
-    extractAuthenticationHeaders(request, `MCC-Authentication`.name, `MCC-Time`.name, false).flatMap { mauthHeader: MauthHeaderValues =>
-      {
-        if (!mauthHeader.authenticator.isEmpty) {
-          provide(mauthHeader)
+  def extractLatestAuthenticationHeaders(disableV1: Boolean): Directive1[MauthHeaderValues] = {
+    extractRequest.flatMap { request: HttpRequest =>
+      // Try to extract and verify V2 headers
+      val authenticator = extractRequestHeader(request, `MCC-Authentication`.name)
+      if (!authenticator.isEmpty) {
+        val timeString = extractRequestHeader(request, `MCC-Time`.name)
+        if (!timeString.isEmpty) {
+          Try(timeString.toLong).toOption match {
+            case Some(time: Long) =>
+              provide(new MauthHeaderValues(authenticator, time))
+            case None => {
+              val msg = s"${`MCC-Time`.name} header supplied with bad format: [$timeString]"
+              logger.error(msg)
+              reject(MalformedHeaderRejection(headerName = `MCC-Time`.name, errorMsg = msg, None))
+            }
+          }
+        } else {
+          reject(MissingHeaderRejection(`MCC-Time`.name))
+        }
+      } else {
+        // If V2 headers not found, fallback to V1 headers if allowed
+        if (!disableV1) {
```

Comment:
Let's avoid double negatives in our code. How about `enableV1`?

Reply:
updated as suggested